pulsar-broker/src/main/java/org/apache/pulsar/broker/service/RedeliveryTracker.java

```diff
@@ -19,17 +19,30 @@
 package org.apache.pulsar.broker.service;
 
 import java.util.List;
+import org.apache.bookkeeper.mledger.Entry;
 import org.apache.bookkeeper.mledger.Position;
 
 public interface RedeliveryTracker {
 
-    int incrementAndGetRedeliveryCount(Position position);
+    int incrementAndGetRedeliveryCount(Position position, Consumer consumer);
 
     int getRedeliveryCount(Position position);
 
-    void remove(Position position);
+    void remove(Position position, Position markDeletedPosition);
 
-    void removeBatch(List<Position> positions);
+    void removeBatch(List<Position> positions, Position markDeletedPosition);
 
     void clear();
+
+    default boolean hasRedeliveredEntry(List<Entry> entries) {
```

Comment:
maybe the default implementation should be no-op, this way we don't add load to users who don't need this feature

Reply:
I feel that since this method exists, it should provide the correct logic, so it should not be `no-op`

Reply:
@codelipenghui what's your take on this ?

Reply:
This method already moved into `InMemoryAndPreventCycleFilterRedeliveryTracker`, see https://github.com/apache/pulsar/pull/17782#discussion_r986623345